docs/proposals/20-ha-control-planes.md

```diff
@@ -528,10 +529,12 @@ In this option existing recovery mechanisms as described above are used. There i
 _Pros:_
 * Less complex to implement since no dynamic re-balancing of pods is required and there is no need to determine if there is an AZ outage.
 * Additional cost to host an HA shoot control plane is kept to the bare minimum.
-* Existing recovery mechanisms are leveraged.
+* Existing recovery mechanisms are leveraged:
+  * `kube-controller-manager` will create new replicas for the affected Deployment Pods and they will recover in short period of time.
+  * The replacement Pods for the affected StatefulSet Pods will fail to be scheduled due to their affinity rules (`etcd` Pods) or volume requirements (`prometheus` and `loki` Pods) to run in the outage zone and will not recover.
 
 _Cons:_
-* Existing recovery of pods will result in a downtime of up to a total of [--machine-health-timeout](https://github.com/gardener/gardener-extension-provider-gcp/blob/dff3d2417ff732fcce69ba10bbe5d04e13781539/charts/internal/machine-controller-manager/seed/templates/deployment.yaml#L49) + [--machine-drain-timeout](https://github.com/gardener/gardener-extension-provider-gcp/blob/dff3d2417ff732fcce69ba10bbe5d04e13781539/charts/internal/machine-controller-manager/seed/templates/deployment.yaml#L48) (which as of today is 2hr10min) for pods which can be rescheduled onto another node/zone. Stateful pods due to the affinity rules will not recover, however stateless pods will eventually recover.
+* Existing recovery of pods will result in a downtime of 5min. For pods that are not part of the Shoot control plane it will be 10min (`--machine-health-timeout`). 
```

Comment:
Where exactly do the `5min` come from, from Gardener's garbage collection? Because then it's only the max downtime, stateless pods usually recover faster, IIUC.

Reply:
Strictly, I wonder if the `5mins` are reflecting the reality.

There are still related KCM properties:
- `--node-monitor-period` (Gardener default `2min`) which makes the node turning into `Unknown`.
- `--pod-eviction-timeout` (Gardener default `2min`) which makes KCM to evict the pod after nodes remain `Unknown` for the configured duration.

Both must be passed until the pod gets a deletion timestamp which marks the `5min` start for the `shoot-care-collector`.

Reply:
> Both must be passed until the pod gets a deletion timestamp which marks the `5min` start for the `shoot-care-collector`.

Yes, nice catch. In general, this garbage collection in the shoot-care-controller was not added as any recovery mechanism and I am really disappointed that I have to document it in this section because it is messing with Terminating Pods and force deleting them after 5min. 😞 

In general, the whole statement did not make much sense. Above we say that Deployment Pods will recover in `short period of time` and in this sentence we try to give concrete downtime times. I reworked the points with the latest force push.

From https://github.com/gardener/gardener/pull/6732#discussion_r997022211, it should be `--node-monitor-grace-period` (2min) and not `--node-monitor-period` (5s).

Wrt to `--pod-eviction-timeout`, according to my testing and reading of KCM's source, `--pod-eviction-timeout` is only used when the taint manager is not enabled. In K8s 1.23 the taint manager is enabled by default and KCM is started with `--enable-taint-manager="true"`. When the taint manager is used, actually the time it takes to evict the affected Pods is `--node-monitor-grace-period` (2min) + 5min = 7min. I don't know yet from where these 5min come from.

For now I adapted it to be `--node-monitor-grace-period` + `--pod-eviction-timeout` but let me investigate from where these 5min come from when the taint manager is used.

Ref https://github.com/kubernetes/kubernetes/blob/v1.23.12/pkg/controller/nodelifecycle/node_lifecycle_controller.go

Reply:
Thank you, @himanshu-kun! It would probably take me a lot of time to find out this. 👍 

I now updated the section describing when KCM starts evicting the affected Pods from unhealthy Nodes.